### PR TITLE
Fix broken API page

### PR DESCRIPTION
**Pull request by aliabid94**
When we changed the documentation style, the view API page for Interfaces got broken. Should be fixed now.

### CODE REVIEW

**Comment by abidlabs** [comment]
What exactly was broken previously @aliabid94?

File: gradio/documentation.py [hunk 1]
```diff
@@ -177,3 +176,16 @@ def generate_documentation():
                 )
             documentation[mode].append(cls_documentation)
     return documentation
+
+
+def document_component_api(component_cls, target):
```
Comment (abidlabs) [inline gradio/documentation.py:181]: Can you give some explanation, what's happening here?

Comment (aliabid94) [reply gradio/documentation.py:181]: Added documentation. Uses

**Comment by aliabid94** [comment]
> What exactly was broken previously @aliabid94?

Previously the app documentation used the docstring to get the types expected by the API. We removed types from the docstring since it was redundant with the type hint. Now "view api" page uses the same type-hint based logic to generate the api page.

File: gradio/components.py [hunk 2]
```diff
@@ -802,9 +782,7 @@ def get_interpretation_scores(
     def generate_sample(self) -> float:
         return self.maximum
 
-        # Output Functionalities
-
-    def postprocess(self, y: float | None) -> float:
+    def postprocess(self, y: float | None) -> float | None:
```
Comment (abidlabs) [inline gradio/components.py:785]: `postprocess()` always produces a float
```suggestion
    def postprocess(self, y: float | None) -> float:
```

**Comment by abidlabs** [comment]
Fixed some typing in some of the components' `preprocess()` and `postprocess()` functions

**Comment by abidlabs** [comment]
Otherwise, LGTM! Nice fix

**Review by abidlabs — APPROVED** [review]

**Pull request merged**